src/main/java/com/cburch/logisim/std/gates/ControlledBuffer.java

```diff
@@ -247,10 +247,10 @@ public void propagate(InstanceState state) {
     final var width = state.getAttributeValue(StdAttr.WIDTH);
     if (control == Value.TRUE) {
       final var in = state.getPortValue(1);
-      if (in == Value.TRUE || in == Value.FALSE) {
+      if (in.isFullyDefined()) {
         state.setPort(0, isInverter ? in.not() : in, GateAttributes.DELAY);
       } else if (in == Value.UNKNOWN) {
-        state.setPort(0, Value.createUnknown(width), GateAttributes.DELAY);
+        state.setPort(0, isInverter ? Value.createError(width) : Value.createUnknown(width), GateAttributes.DELAY);
       } else if (in == Value.ERROR) {
         state.setPort(0, Value.createError(width), GateAttributes.DELAY);
       }
```

Comment:
That still doesn't work. Value.UNKNOWN and Value.ERROR are one-bit values. They will not be equal to any value with more than one bit. isFullyDefined() means no errors or unknowns. Any multi-bit value may have some of each of the 4 values. Each bit should be processed independently. So no test on `in` makes sense.

All you need is the line:
`state.setPort(0, isInverter ? in.not() : in, GateAttributes.DELAY);`
That will pass through all the original values for the buffer and negate 0/1 and make U & E become E for the inverter.